extension/chrome/settings/modules/contacts.ts

```diff
@@ -23,6 +33,48 @@ Catch.try(async () => {
 
   BrowserMsg.listen(tabId); // set_css
 
+  const attUI = new AttUI(() => Promise.resolve({ size_mb: 5, size: 5 * 1024 * 1024, count: 1 }));
+  attUI.initAttDialog('fineuploader', 'fineuploader_button');
+  attUI.setAttAddedCb(async (file: Att) => {
+    (file !== null && typeof file !== 'undefined' && typeof file.id !== 'undefined' && file.id !== '') ? attUI.clearAtt(file.id) : attUI.clearAtts();
```

Comment:
Actually, does clearing / canceling attachments serve any purpose here? Does the functionality change when you remove it?

Reply:
What was happening, if I decided I wanted to change the file imported (picked the wrong file, or a file that didn't have any PGP keys) clicking the import button again it would throw an error modal about having too many attachments. So since I am not actually uploading the attachments, I thought I could clear the attachment variable as soon as I had the file object in my callback.

![image](https://user-images.githubusercontent.com/17385251/54480753-08fd8180-47f2-11e9-9e7c-53f45f3ab286.png)

Reply:
Got it, makes sense.

Reply:
In this case you should be able to just use `cancelAtt` (made public) with the id without all the checks in front of it (if I'm not wrong).

Reply:
Slight problem with the above, the `attUI.setAttAddedCb` is returning a file object with an 'undefined' id field. This is the object I am having returned.

```
{
    bytes: Uint8Array[.......],
    cid: undefined,
    getData: () => {....},
    hasData: () => {....},
    id: undefined,
    inline: true,
    length: 27115,
    msgId: undefined,
    name: "pbr.asc",
    setData: (bytes) => {....},
    treatAs: () => {....},
    treatAsValue: undefined,
    type: "text/plain",
    url: undefined
}
```

Reply:
Got it. You have found an api where the naming is not the best(on a second look), because the attachment id represents a different thing than the file upload id.

In this case, the simplest thing is to do `clearAtts` as you did, but I would call it `cancelAllAtts()` to resemble original naming, and make it public. Then `cancelAtt(id)` would stay private.

Reply:
Having you work on this uncovers some issues with my own code that's not so clear, I'll make an issue for this.